lib/openssl.ml

```diff
@@ -345,10 +345,20 @@ let connect ~hostname ~config ~alpn_protocols fd =
       in
       let s = Lwt_ssl.embed_uninitialized_socket fd ctx in
       let ssl_sock = Lwt_ssl.ssl_socket_of_uninitialized_socket s in
-      Ssl.set_client_SNI_hostname ssl_sock hostname;
-      (* https://wiki.openssl.org/index.php/Hostname_validation *)
-      Ssl.set_hostflags ssl_sock [ No_partial_wildcards ];
-      Ssl.set_host ssl_sock hostname;
+      (* If hostname is an IP address, check that instead of the hostname *)
+      let ipaddr = Ipaddr.of_string hostname in
+      let* () =
```

Comment:
why is this `let*` necessary? all the calls in this block seem synchronous.

Reply:
This might be me being fairly new to Ocaml, but if I take the pattern match out of the block the compiler complains with:

```
File "lib/openssl.ml", line 353, characters 8-52:
353 |         Ssl.set_ip ssl_sock (Ipaddr.to_string ipadr);
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: This expression has type unit but an expression was expected of type
         ('a, [> `Connect_error of string | `Exn of exn ] as 'b)
         Piaf__Monads.Lwt_result.t = ('a, 'b) result Lwt.t
```